sentry_sdk/scope.py

```diff
@@ -427,9 +534,10 @@ def generate_propagation_context(self, incoming_data=None):
                     self._propagation_context,
                 )
 
-        if self._propagation_context is None:
+        if self._propagation_context is None and self._type != ScopeType.CURRENT:
```

Comment:
How does the `propagation_context` now work in general wrt scopes? Why the `_type` check here?